### PR TITLE
Visual Studio Tools for Docker How-To

### PR DESCRIPTION
**Pull request by spboyer**
# Title

Using Visual Studio Tools for Docker 
## Summary

Introduction to Visual Studio Tools for Docker 

Developing and debugging your application in a Docker container can be a ceremony of tasks to get setup with various tools. [Visual Studio Tools for Docker](https://visualstudiogallery.msdn.microsoft.com/0f5b2caa-ea00-41c8-b8a2-058c7da0b3e4) helps you get past the hurdles and find the bugs using F5 to debug your application directly in a locally hosted Docker Container. 

More content for #806 
## Suggested Reviewers

@stevelasker @glennc @BillWagner

### CODE REVIEW

**Comment by qinezh** [comment]
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/core/docker/visual-studio-tools-for-docker?branch=pr-en-us-1035](https://review.docs.microsoft.com/en-us/dotnet/articles/core/docker/visual-studio-tools-for-docker?branch=pr-en-us-1035)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C9%5C19%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201609191644541542-1035%5Cworkflow_report.html).

File: docs/core/docker/visual-studio-tools-for-docker.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+
+# Visual Studio Tools for Docker 
+Developing and debugging your application in a Docker container can be a ceremony of tasks to get setup with various tools. [Visual Studio Tools for Docker](https://visualstudiogallery.msdn.microsoft.com/0f5b2caa-ea00-41c8-b8a2-058c7da0b3e4) helps you get past the hurdles and find the bugs using F5 to debug your application directly in a locally hosted Docker Container. 
+
+> [!NOTE]
+>The current version targets Linux Docker containers, with Windows Containers coming soon.
+
+## Prerequisites
+- [Microsoft Visual Studio 2015 Update 3](https://www.visualstudio.com/downloads/download-visual-studio-vs)
+- [Microsoft .NET Core 1.0.1 RTM VS 2015 Tooling Preview 2, with the 1.0.1 updates](https://go.microsoft.com/fwlink/?LinkID=827546)
```
Comment (mairaw) [inline docs/core/docker/visual-studio-tools-for-docker.md]: do you need to say this "with the 1.0.1 updates"?

File: docs/core/docker/visual-studio-tools-for-docker.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+Project Context Menu
+
+![Right Click Add Docker Support](./media/visual-studio-tools-for-docker/right-click-add-docker-support.png)
+
+The following files are added the to the project.
+
+- **Dockerfile**: the Docker file for ASP.NET Core applications is based on the [microsoft/aspnetcore](https://hub.docker.com/r/microsoft/aspnetcore) image. This image includes the ASP.NET Core NuGet packages, which have been pre-jitted improving startup performance. When building .NET Core Console Applications, the Dockerfile FROM will reference the [microsoft/dotnet:1.0.1](https://hub.docker.com/r/microsoft/dotnet) image.
+- **docker-compose.yml**: base Docker Compose file
+- **docker-compose.dev.debug.yml**: Docker Compose file with additional settings for ASPNETCORE_ENVIRONMENT, volume mapping and ---
+title: Visual Studio Tools for Docker
```
Comment (mairaw) [inline docs/core/docker/visual-studio-tools-for-docker.md]: there's something odd with this topic. You restart the topic several times with metadata sections in the middle of the text.

**Review by mairaw — CHANGES_REQUESTED** [review]
The new topic is broken. Please fix that and then we can proceed with the review of this topic.

**Comment by spboyer** [comment]
@mairaw sorry about that, local merge must have gotten screwy. Fixed.

**Comment by mairaw** [comment]
No worries @spboyer. It was just hard to read like that. :smile:

**Comment by qinezh** [comment]
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/core/docker/visual-studio-tools-for-docker?branch=pr-en-us-1035](https://review.docs.microsoft.com/en-us/dotnet/articles/core/docker/visual-studio-tools-for-docker?branch=pr-en-us-1035)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C9%5C19%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201609191759470990-1035%5Cworkflow_report.html).

File: docs/core/docker/visual-studio-tools-for-docker.md (hunk 1 above)
Comment (spboyer) [inline docs/core/docker/visual-studio-tools-for-docker.md]: I dont think that's overly important, but up to @stevelasker to comment

Comment (mairaw) [inline docs/core/docker/visual-studio-tools-for-docker.md]: I was mostly thinking about the right branding which would be ".NET Core 1.0.1 - VS 2015 Tooling Preview 2".

File: docs/core/docker/visual-studio-tools-for-docker.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+A required configuration is to setup **[Shared Drives](https://docs.docker.com/docker-for-windows/#/shared-drives)** in Docker for Windows. The setting is required for the volume mapping and debugging support.
+
+Right click the Docker icon in the System Tray, click Settings and select Shared Drives.
+
+![Shared Drives](./media/visual-studio-tools-for-docker/settings-shared-drives-win.png) 
+
+## Create an ASP.NET Web Application and add Docker Support
+
+Using Visual Studio, create a new ASP.NET Core Web Application. When the application is loaded, either select **Add Docker Support** from the **Project Menu** or right click the project from the Solution Explorer and select Add > Docker Support.
```
Comment (mairaw) [inline docs/core/docker/visual-studio-tools-for-docker.md]: nit: Add and Docker Support in the end should also be bold.

File: docs/core/docker/visual-studio-tools-for-docker.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+> [!NOTE]
+>The current version targets Linux Docker containers, with Windows Containers coming soon.
+
+## Prerequisites
+- [Microsoft Visual Studio 2015 Update 3](https://www.visualstudio.com/downloads/download-visual-studio-vs)
+- [Microsoft .NET Core 1.0.1 RTM VS 2015 Tooling Preview 2, with the 1.0.1 updates](https://go.microsoft.com/fwlink/?LinkID=827546)
+- [Docker For Windows](https://www.docker.com/products/docker#/windows) to run your Docker containers locally
+
+## Installation and setup
+Download and install the [Visual Studio Tools for Docker](https://visualstudiogallery.msdn.microsoft.com/0f5b2caa-ea00-41c8-b8a2-058c7da0b3e4) from the [Visual Studio Gallery](http://visualstudiogallery.msdn.microsoft.com/).
```
Comment (BillWagner) [inline docs/core/docker/visual-studio-tools-for-docker.md]: You can also search for it in the Extensions window in VS, right?

File: docs/core/docker/visual-studio-tools-for-docker.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+Project Menu
+
+![Project Add Docker Support](./media/visual-studio-tools-for-docker/project-add-docker-support.png)
+
+Project Context Menu
+
+![Right Click Add Docker Support](./media/visual-studio-tools-for-docker/right-click-add-docker-support.png)
+
+The following files are added the to the project.
```
Comment (BillWagner) [inline docs/core/docker/visual-studio-tools-for-docker.md]: typo: "added the to the" -> "added to the"

File: docs/core/docker/visual-studio-tools-for-docker.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+- **Dockerfile**: the Docker file for ASP.NET Core applications is based on the [microsoft/aspnetcore](https://hub.docker.com/r/microsoft/aspnetcore) image. This image includes the ASP.NET Core NuGet packages, which have been pre-jitted improving startup performance. When building .NET Core Console Applications, the Dockerfile FROM will reference the [microsoft/dotnet:1.0.1](https://hub.docker.com/r/microsoft/dotnet) image.
```
Comment (BillWagner) [inline docs/core/docker/visual-studio-tools-for-docker.md]: Would it be better to say that the FROM will reference the most recent version? That would make it more resilient to nwe .NET Core versions.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this is great. I commented on a typo or two to fix, and a couple content suggestions.

File: docs/core/docker/visual-studio-tools-for-docker.md (hunk 4 above)
Comment (spboyer) [inline docs/core/docker/visual-studio-tools-for-docker.md]: Added

File: .gitignore [hunk 7]
```diff
@@ -215,3 +215,4 @@ _themes/
 log/
 .optemp/
 .openpublishing.buildcore.ps1
+.vscode/
```
Comment (spboyer) [inline .gitignore]: ok, will remove this

File: docs/core/docker/visual-studio-tools-for-docker.md (hunk 5 above)
Comment (spboyer) [inline docs/core/docker/visual-studio-tools-for-docker.md]: fixed

File: docs/core/docker/visual-studio-tools-for-docker.md (hunk 6 above)
Comment (spboyer) [inline docs/core/docker/visual-studio-tools-for-docker.md]: sure, makes sense.

File: docs/core/docker/visual-studio-tools-for-docker.md (hunk 1 above)
Comment (spboyer) [inline docs/core/docker/visual-studio-tools-for-docker.md]: Made the change.

File: docs/core/docker/visual-studio-tools-for-docker.md (hunk 3 above)
Comment (spboyer) [inline docs/core/docker/visual-studio-tools-for-docker.md]: fixed

**Comment by spboyer** [comment]
@mairaw @BillWagner - fixed nits

**Comment by qinezh** [comment]
Open Publishing Build Service: The pull request content has been published and here are some changed files links of this time:
- Docset 1 - **core-docs**  
  [https://review.docs.microsoft.com/en-us/dotnet/articles/core/docker/visual-studio-tools-for-docker?branch=pr-en-us-1035](https://review.docs.microsoft.com/en-us/dotnet/articles/core/docker/visual-studio-tools-for-docker?branch=pr-en-us-1035)  

**Status**: Succeeded.  
[Open Publishing Report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C9%5C20%5C2a032abb-a76c-bfc8-4231-71ab612989a2%5CPullRequest%5C201609201217511028-1035%5Cworkflow_report.html).

**Review by BillWagner — APPROVED** [review]

**Review by mairaw — APPROVED** [review]

**Pull request merged**